components/fxa-client/src/error.rs

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use thiserror::Error;
+
+/// Generic error type thrown by many [`FirefoxAccount`] operations.
+///
+/// Precise details of the error are hidden from consumers, mostly due to limitations of
+/// how we expose this API to other languages. The type of the error indicates how the
+/// calling code should respond.
```

Comment:
I don't think we have the same UniFFI limitations anymore, should we be adding more data to these?  Or is it actually preferable to hide them?

Reply:
I would switch us to use the error handling model we introduced with logins and places, I like how simple the errors at this layer are (they make me think that `FxaError` sounds like a nice `ExternalError`!)

Reply:
I agree, only errors which would ever be actually caught should be distinguished, the rest lumped together and reported via error reporting. I'm not sure this PR captures that correctly though, but not sure if it is the intention - specifically, I'd never expect consumers to catch `Panic` specifically, but could imagine it *would* want to handle `BackoffError` (or maybe more generally a "server error"?)

Looking lower, maybe this PR *is* attempting to get those errors correct - if so I think it needs changes.

Reply:
Good point.  I think we can get rid of `NoExistingAuthFlow`, `WrongAuthFlow`, and `Panic`, then maybe rename the remaining to `AuthError`, `NetworkError`, and `UnexpectedError`.

But I'm thinking we can get this one merged now and work on the breaking changes in the next pass.